### PR TITLE
Add VM instance create time property for all providers

### PR DESCRIPTION
**Pull request by rodrigonull**
Add the ability to retrieve the VM instance creation time for all supported providers.

### CODE REVIEW

File: tests/test_compute_service.py [hunk 1]
```diff
@@ -168,6 +168,9 @@ def test_instance_properties(self):
             self.assertEqual(len(find_zone), 1,
                              "Instance's placement zone could not be "
                              " found in zones list")
+            self.assertIsNotNone(
+                test_instance.create_time,
+                "Instance must have it's creation time")
```
Comment (nuwang) [inline tests/test_compute_service.py:173]:
```suggestion
                "Instance must have its creation time")
```

Comment (nuwang) [reply tests/test_compute_service.py:173]: Can we also add a test here to make sure that the returned object is always a python datetime object?

**Review by nuwang — COMMENTED** [review]
@rodrigonull Thanks for this contribution. Do all the methods return a python datetime object? Can we add a test for that?

**Comment by rodrigonull** [comment]
> @rodrigonull Thanks for this contribution. Do all the methods return a python datetime object? Can we add a test for that?

Only AWS and Azure return a datetime object. GCP and OpenStack return an ISO 8601 string. I was wondering if we should convert the string to datetime, but that probably would require to import a third party library, like [dateutil](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse), because I think only in python 3.7 a [method](https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat) was added that allows to convert a ISO 8601 string to datetime. What do you think?

**Comment by nuwang** [comment]
I think only in python 3.7 a [method](https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat) was added that allows to convert a ISO 8601 string to datetime. What do you think?

I think we could go for that, since support for python<3 has already been dropped, and most people are likely above 3.7. Alternatively, we could return an ISO 8601 string if that's easier.

**Review by nuwang — APPROVED** [review]
Thanks, this looks good!

**Comment by nuwang** [comment]
Released in v3.1.0

**Pull request merged**